src/lib/y2partitioner/widgets/encrypt_password.rb

```diff
@@ -93,6 +73,94 @@ def pw1
       def pw2
         Yast::UI.QueryWidget(Id(:pw2), :Value)
       end
+
+      # @return [Boolean]
+      def random_password_allowed?
+        @random_password_allowed ||= @controller.filesystem_type.to_sym == :swap
+      end
+
+      # @return [Boolean]
+      def random_password?
+        return false unless random_password_allowed?
+
+        Yast::UI::QueryWidget(Id(:pwd_type), :Value) == :random_pwd
+      end
+
+      def form_content
+        if random_password_allowed?
+          extended_content
+        else
+          password_fields
+        end
+      end
+
+      def extended_content
+        VBox(
+          RadioButtonGroup(
+            Id(:pwd_type),
+            VBox(
+              Left(RadioButton(Id(:manual_pwd), _("Ask for password on boot to mount the swap"), true)),
+              password_fields,
+              Left(RadioButton(Id(:random_pwd), _("Auto-generate a new &random password on every boot"))),
+            )
+          )
+        )
+      end
+
+      def password_fields
+        MarginBox(
+          1.45,
+          0.5,
+          VBox(
+            Password(
+              Id(:pw1),
+              Opt(:hstretch),
+              # Label: get password for user root
+              # Please use newline if label is longer than 40 characters
+              _("&Enter a Password for your File System:"),
+              ""
+            ),
+            Password(
+              Id(:pw2),
+              Opt(:hstretch),
+              # Label: get same password again for verification
+              # Please use newline if label is longer than 40 characters
+              _("Reenter the Password for &Verification:"),
+              ""
+            ),
+            VSpacing(0.5)
+          )
+        )
+      end
+
+      # @return [String]
+      def help_encryption_password
+        if random_password_allowed?
+          _(
+            "<p>\n" \
+              "You will need to enter your encryption password or allow to system\n" \
+              "auto-generate a new random password on every boot.\n" \
+            "</p>\n"
+          )
+        else
+        _(
+          "<p>\n" \
+            "You will need to enter your encryption password.\n" \
+          "</p>\n"
+        )
+        end
+      end
+
+      # @return [String]
+      def help_skip
+        _(
+          "<p>\n" \
+            "If the encrypted file system does not contain any system file and therefore is\n" \
+            "not needed for the update, you may select <b>Skip</b>. In this case, the\n" \
+            "file system is not accessed during update.\n" \
+          "</p>\n"
```

Comment:
to be honest now I am lost. This widget talk about setting encrypted password and now it mention some update. Relict of old help probably? But quite confusing for me.

Reply:
I also do not see that skip option ( unless I overlook it ).

Reply:
> Relict of old help probably

I guess, because that already was there. I only added a clarification about random password option.

In fact, the subject of [commit which added](https://github.com/yast/yast-storage-ng/commit/49731c72f6ee5415ddbd095b1d517c397523137b) it is 

` Bulk copy of yast-partitioner files with no adaptation `

Shall I take the opportunity and remove it?